diff --git a/Common/Bus.cpp b/Common/Bus.cpp
index 80875d5..0a34e59 100644
--- a/Common/Bus.cpp
+++ b/Common/Bus.cpp
@@ -36,9 +36,9 @@ namespace Bus
         Nodes.resize( Allocation );
         RefNo.resize( Allocation );
         FNumNodesThisBus = 0;
-        Ysc = NULL;
-        Zsc = NULL;
-        Zsc012 = NULL;
+        Ysc = 0;
+        Zsc = 0;
+        Zsc012 = 0;
         VBus.clear();
         BusCurrent.clear();
         kVBase = 0.0;  // Signify that it has not been set
diff --git a/Common/DSSGlobals.cpp b/Common/DSSGlobals.cpp
index 1353ba3..4365ad8 100644
--- a/Common/DSSGlobals.cpp
+++ b/Common/DSSGlobals.cpp
@@ -1335,8 +1335,8 @@ namespace DSSGlobals
         FM_MHandle[ActiveActor] = NULL;
         DIFilesAreOpen[ActiveActor] = false;
         ActiveVSource[ActiveActor] = NULL;
-        DSSObjs[ActiveActor] = NULL;
-        DSSClassList[ActiveActor] = NULL;
+        DSSObjs[ActiveActor] = 0;
+        DSSClassList[ActiveActor] = 0;
       }
 
       AutoDisplayShowReport = true;
diff --git a/Support/Sysutils.cpp b/Support/Sysutils.cpp
index 99c87f8..009f735 100644
--- a/Support/Sysutils.cpp
+++ b/Support/Sysutils.cpp
@@ -8673,7 +8673,7 @@ bool RemoveDir( const String& Dir )
                      << int ( 0 ) << int ( 1 ) )
 
 
-
+#ifndef ABDAY_1
 const int ABDAY_1 = ( __LC_TIME << 16 );
 const int DAY_1 = ( ABDAY_1 ) + 7;
 const int ABMON_1 = ( ABDAY_1 ) + 14;
@@ -8684,9 +8684,18 @@ const int D_T_FMT = ( ABDAY_1 ) + 40;
 const int D_FMT = ( ABDAY_1 ) + 41;
 const int T_FMT = ( ABDAY_1 ) + 42;
 const int T_FMT_AMPM = ( ABDAY_1 ) + 43;
+#endif
+#ifndef __LC_NUMERIC
+#define __LC_NUMERIC LC_NUMERIC
+#endif
 const int __DECIMAL_POINT = ( __LC_NUMERIC << 16 );
+#ifndef RADIXCHAR
 const int RADIXCHAR = __DECIMAL_POINT;
+#endif
 const int __THOUSANDS_SEP = ( __DECIMAL_POINT ) + 1;
+#ifndef __LC_MONETARY
+#define __LC_MONETARY LC_MONETARY
+#endif
 const int __INT_CURR_SYMBOL = ( __LC_MONETARY << 16 );
 const int __CURRENCY_SYMBOL = ( __INT_CURR_SYMBOL ) + 1;
 const int __MON_DECIMAL_POINT = ( __INT_CURR_SYMBOL ) + 2;
@@ -8921,7 +8930,7 @@ void GetFormatSettings( )
 
   int i = 0;
   unsigned char prec = '\0', sep = '\0', signp = '\0';
-  setlocale( __LC_ALL, "" );
+  setlocale( LC_ALL, "" );
   for ( i = 1; i <= 12; i++)
   {
     ShortMonthNames[i - 1] = GetLocaleStr( ABMON_1 + i - 1 );
diff --git a/Support/d2c_sysexcept.cpp b/Support/d2c_sysexcept.cpp
index fcf90ed..de9ea01 100644
--- a/Support/d2c_sysexcept.cpp
+++ b/Support/d2c_sysexcept.cpp
@@ -78,7 +78,7 @@
 #include <stdio.h>
 
 #ifdef linux
-#include <error.h>
+#include <err.h>
 #endif
 
 using namespace std;
